src/test/denialofservice_tests.cpp

```diff
@@ -63,10 +114,9 @@ BOOST_AUTO_TEST_CASE(outbound_slow_chain_eviction)
                      /*addrNameIn=*/"",
                      ConnectionType::OUTBOUND_FULL_RELAY,
                      /*inbound_onion=*/false};
-    dummyNode1.SetCommonVersion(PROTOCOL_VERSION);
 
-    peerLogic->InitializeNode(&dummyNode1, ServiceFlags(NODE_NETWORK|NODE_WITNESS));
-    dummyNode1.fSuccessfullyConnected = true;
+    peerman.InitializeNode(&dummyNode1, ServiceFlags(NODE_NETWORK|NODE_WITNESS));
+    ConnectNode(connman, peerman, dummyNode1, ServiceFlags(NODE_NETWORK|NODE_WITNESS));
```

Comment:
This is done because this PR removes `CNode::fClient` and `CNode::m_limited_node` members. These two variables are default initialized as `false` and then set to the correct value when a `version` message is received. In production this is fine, as the two bools are not used if the version handshake hasn't happened. In the test however, we don't do the version handshake and manually set `fSuccessfullyConnected = true`. The default `false` then results in the logic behaving as expected (e.g. the node is not a client, so we ask them for headers).

Now with the PR the two bools are removed and replaced with static functions that operate on `Peer::their_services` to determine client and pruned node status. In the test here, if we keep `Peer::their_services` as `NODE_NONE` then we would consider `dummyNode1` a client and not ask for headers.

I think this makes the test more realistic (also mentioned in the commit message) and shouldn't change behavior.